File: Code/dotNetBasics/ConsoleApp/ADStack/Components/PedestrianDetector.cs
using ConsoleApp.ADStack.Models;

namespace ConsoleApp.ADStack.Components
{
    /// <summary>
    /// Fusion module for pedestrian detector
    /// </summary>
    internal class PedestrianDetector : IFusion
    {
        public void Bind(IComponent component)
        {
            Console.WriteLine($"[PedestrianDetector] Binding component: {component.GetType().Name}");
        }

        public void Start()
        {
            Console.WriteLine("[PedestrianDetector] Starting.");
        }

        public void Stop()
        {
            Console.WriteLine("[PedestrianDetector] Stopping.");
        }
    }
}

File: Code/Fluent.Java/src/adstack/components/PedestrianDetector.java
package adstack.components;

import adstack.models.IComponent;
import adstack.models.IFusion;

public class PedestrianDetector implements IFusion {
	
	@Override
	public void bind(IComponent component) {
        System.out.println("[PedestrianDetector] Binding component: " + component.getClass().getName());
	}
	
	@Override
	public void start() {
        System.out.println("[PedestrianDetector] Starting.");
	}

	@Override
	public void stop() {
        System.out.println("[PedestrianDetector] Stopping.");
	}

}
